### PR TITLE
18 frontend authenticate flow

### PR DESCRIPTION
**Pull request by sarunnut1236**
- Enable middleware protecting unauthenticated users

This is proof of work
[https://jam.dev/c/a37aaba1-e559-4128-82d1-580952e06e97](https://jam.dev/c/a37aaba1-e559-4128-82d1-580952e06e97)

### CODE REVIEW

File: frontend/public/icon/google.svg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" xmlns:xlink="http://www.w3.org/1999/xlink" style="display: block;">
```
Comment (viruskizz) [inline frontend/public/icon/google.svg:1]: Please move this file to
frontend/assets/icons/...

File: frontend/src/app/app.routes.ts [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+import { LoginTestComponent } from './features/test/login-test/login-test.component';
+import { MemberTestComponent } from './features/test/member-test/member-test.component';
+import { TestComponent } from './features/test/test.component';
+import { PageNotFoundComponent } from './shared/page-not-found/page-not-found.component';
+import { AuthGuard } from './auth.guard';
 
 export const routes: Routes = [
   {
     path: '',
     component: HomeComponent,
```
Comment (viruskizz) [inline frontend/src/app/app.routes.ts:12]: Do we need to add pathMatch: full ?

File: frontend/src/app/auth.guard.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        } else {
+          this.router.navigate(['/test/login']);
+          throw new Error('Token not found');
+        }
+      });
+    } catch (error) {
+      this.router.navigate(['/test/login']);
+      console.error('AuthGuard.canActivate', error);
+      return false;
+    }
```
Comment (viruskizz) [inline frontend/src/app/auth.guard.ts:38]: Improve: to pipe ?

return firebaseUser.pipe(
switchMap(),
catchError(),
)

You can see more concept at intercepters file in backend path

File: frontend/src/app/app.routes.ts (hunk 2 above)
Comment (sarunnut1236) [reply frontend/src/app/app.routes.ts:13]: added

File: frontend/public/icon/google.svg (hunk 1 above)
Comment (sarunnut1236) [reply frontend/public/icon/google.svg]: done

File: frontend/src/app/auth.guard.ts (hunk 3 above)
Comment (sarunnut1236) [reply frontend/src/app/auth.guard.ts]: Edited krub

**Comment by viruskizz** [comment]
LGTM

**Review by viruskizz — APPROVED** [review]

**Pull request merged**